demos/signature/EVP_Signature_demo.c

```diff
@@ -0,0 +1,238 @@
+/*-
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* An example that uses the EVP_MD*, EVP_DigestSign* and EVP_DigestVerify* 
+ * methods to calculate and verify a signature of two static buffers.
+ * Another way of generating a signature is to use EVP_PKEY_sign(), it uses
+ * a digest of message, does not hash the data to be signed, and therefore
+ * is normally used to sign digests.
+ */
+
+#include <string.h>
+#include <stdio.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include "EVP_Signature_demo.h"
+#include <openssl/decoder.h>
+
+/*
+ * This demonstration will calculate and verify a signature of data using
+ * the soliloqy from Hamlet scene 1 act 3
+ */
+
+static const char *hamlet_1 =
+    "To be, or not to be, that is the question,\n"
+    "Whether tis nobler in the minde to suffer\n"
+    "The slings and arrowes of outragious fortune,\n"
+    "Or to take Armes again in a sea of troubles,\n"
+;
+static const char *hamlet_2 =
+    "And by opposing, end them, to die to sleep;\n"
+    "No more, and by a sleep, to say we end\n"
+    "The heart-ache, and the thousand natural shocks\n"
+    "That flesh is heir to? tis a consumation\n"
+;
+
+/*
+ * For demo_sign, load EC private key priv_key from priv_key_der[].
+ * For demo_verify, load EC public key pub_key from pub_key_der[].
+ */
+static EVP_PKEY *get_key(OSSL_LIB_CTX *libctx, const char *propq, int public)
+{
+    OSSL_DECODER_CTX *dctx = NULL;
+    EVP_PKEY  *pkey = NULL;
+    int selection;
+    const unsigned char *data;
+    size_t data_len;
+
+    if (public) {
+        selection = EVP_PKEY_PUBLIC_KEY;
+        data =  pub_key_der;
+        data_len = sizeof(pub_key_der);
+    } else {
+        selection =  EVP_PKEY_KEYPAIR;
+        data = priv_key_der;
+        data_len = sizeof(priv_key_der);
+    }
+    dctx = OSSL_DECODER_CTX_new_for_pkey(&pkey, "DER", NULL, "EC",
+                                         selection, libctx, propq);
+    (void)OSSL_DECODER_from_data(dctx, &data, &data_len);
+    OSSL_DECODER_CTX_free(dctx);
+    if (pkey == NULL)
+        fprintf(stderr, "Failed to load %s key.\n", public ? "Public" : "Private");
+    return pkey;
+}
+
+static int demo_sign(OSSL_LIB_CTX *libctx,  const char *sig_name,
+                     size_t *sig_out_len, unsigned char **sig_out_value)
+{
+    int i = 0, j = 0, result = 0, public = 0;
```

Comment:
i is unused

Reply:
Done, thanks for your comments. 
Also removed unused _i_ in demo_verify.